homeassistant/components/zha/__init__.py

```diff
@@ -299,6 +307,8 @@ def register_entity(self, ieee, entity_obj):
             'entity_suffix': '_{}'.format(cluster.cluster_id),
         }
         discovery_info[discovery_attr] = {cluster.cluster_id: cluster}
+        if sub_component:
+            discovery_info.update({'sub_component': sub_component})
```

Comment:
Discovery info is send over the event bus and thus needs to be JSON serializable. Cannot be classes.

Reply:
@balloob  `discovery_info` is stored in `hass.data`. We only send the key over the event bus and then use the key to retrieve `discovery_info` from `hass.data` in `async_setup_platform()`.